entry.js

```diff
@@ -10,9 +12,23 @@ function Entry(relativePath, basePath, mode, size, mtime) {
 }
 
 
-Entry.prototype.isDirectory = function() {
+Entry.prototype.isDirectory = function isDirectory() {
   /*jshint -W016 */
   return (this.mode & 61440) === 16384;
 };
 
+/**
+  On systems where we cannot symlink, do not add any special inequality cases
+  (ie always return true here, and only get `change` operations for files with
+  different `mtime`, `size` or `mode`).
+
+  On systems where we can symlink, also count a directory whose `linkDir` has
+  changed as different.  This allows us to handle the case where a directory
+  goes from being copied (existing in multiple inputPaths) to being symlinked
+  (existing in only one inputPath) or vice-versa.
+*/
+Entry.prototype.equals = function equals(otherEntry) {
+  return !canSymlink || (this.linkDir === otherEntry.linkDir);
```

Comment:
> Im wondering if we should handle the other comparisons here as well for files

I wouldn't want to do that unless we can find a nice way of not requiring this for consumers who who don't need custom equality comparison at all.

I guess we could provide a "base class" `require('fs-tree-diff).entry` and allow consumers to "subclass" (in that prototypal way) if they want a custom `equals`?

Then the merge-trees `entry` becomes something like

```js
var BaseEntry = require('fs-tree-diff').entry;

function Entry(relativePath, basePath, mode, size, mtime) {
  BaseEntry.call(this, relativePath, mode, size, mtime);

  this.basePath = basePath;
  this.linkDir = false;
}

Entry.prototype.equals = function equals(otherEntry) {
  return BaseEntry.prototype.equals(otherEntry) && this._symlinkEquals(otherEntry);
}

Entry.prototype._symlinkEquals = function symlinkEquals(otherEntry) {
  return !canSymlink || (this.linkDir === otherEntry.linkDir);
}
```

thoughts?


> and the existing code you have in this function is executing in the isDirectory() branch

Yes this is intentional for readability to make the cases more clear (rather than relying on knowing that certain cases won't happen because of the way `equals` works).

Do you think I should strip them and just leave an explanatory comment instead?